src/diagnostics/MOM_diagnose_KdWork.F90

```diff
@@ -0,0 +1,1007 @@
+! consistency testing. These are noted in comments with units like Z, H, L, and T, along with
+! their mks counterparts with notation like "a velocity [Z T-1 ~> m s-1]".  If the units
+! vary with the Boussinesq approximation, the Boussinesq variant is given first.
+
+contains
+
+!> Loop over all implemented diffusivities to diagnose and output Kd Work/buoyancy fluxes
+subroutine KdWork_Diagnostics(G,GV,US,diag,VBF,N2_Salt,N2_Temp,dz)
+  type(ocean_grid_type),      intent(in)    :: G       !< Grid type
+  type(verticalGrid_type),    intent(in)    :: GV      !< ocean vertical grid structure
+  type(unit_scale_type),      intent(in)    :: US      !< A dimensional unit scaling type
+  type(diag_ctrl), target,    intent(inout) :: diag    !< regulates diagnostic output
+  type (vbf_CS),              intent(inout) :: VBF     !< Vertical buoyancy flux structure
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), &
+                              intent(in)    :: N2_Salt !< Buoyancy frequency [T-2 ~> s-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)+1), &
+                              intent(in)    :: N2_Temp !< Buoyancy frequency [T-2 ~> s-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), &
+                              intent(in)    :: dz      !< Grid spacing [Z ~> m]
+
+  ! Work arrays for computing buoyancy flux integrals
+  real, dimension(SZI_(G),SZJ_(G)) :: work2d, work2d_salt, work2d_temp
+  real :: work, work_salt, work_temp
+
+  integer :: k, nz
+
+  nz = GV%ke
+
+  ! Compute total fluxes
+  if (VBF%id_Bdif_dz>0 .or. VBF%id_Bdif_salt_dz>0 .or. VBF%id_Bdif_temp_dz>0 .or. &
+      VBF%id_Bdif_idz>0 .or. VBF%id_Bdif_salt_idz>0 .or. VBF%id_Bdif_temp_idz>0 .or. &
+      VBF%id_Bdif_idV>0 .or. VBF%id_Bdif_salt_idV>0 .or. VBF%id_Bdif_temp_idV>0 ) then ! Doing vertical integrals
+    ! Do Salt
+    if (VBF%id_Bdif_salt_dz>0 .or. VBF%id_Bdif_dz>0 .or. VBF%id_Bdif_salt>0 .or. VBF%id_Bdif>0 .or. &
+        VBF%id_Bdif_idz>0 .or. VBF%id_Bdif_salt_idz>0 .or. VBF%id_Bdif_idV>0 .or. VBF%id_Bdif_salt_idV>0) &
+      call diagnoseKdWork(G, GV, N2_salt, VBF%Kd_salt, VBF%Bflx_salt, dz = dz, Bdif_flx_dz=VBF%Bflx_salt_dz)
+    ! Do Temp
+    if (VBF%id_Bdif_temp_dz>0 .or. VBF%id_Bdif_dz>0 .or. VBF%id_Bdif_temp>0 .or. VBF%id_Bdif>0 .or. &
+        VBF%id_Bdif_idz>0 .or. VBF%id_Bdif_temp_idz>0 .or. VBF%id_Bdif_idV>0 .or. VBF%id_Bdif_temp_idV>0) &
+       call diagnoseKdWork(G, GV, N2_temp, VBF%Kd_temp, VBF%Bflx_temp, dz = dz, Bdif_flx_dz=VBF%Bflx_temp_dz)
+    if (VBF%id_Bdif_temp_idz>0 .or. VBF%id_Bdif_idz>0) then
+      work2d_temp(:,:) = 0.0
+      do k = 1,nz
+        work2d_temp(:,:) = work2d_temp(:,:) + VBF%Bflx_temp_dz(:,:,k)
+      enddo
+    endif
+    if (VBF%id_Bdif_temp_idV>0 .or. VBF%id_Bdif_idV>0) then
+      work_temp = 0.0
+      do k = 1,nz
+        work_temp = work_temp + global_area_integral(VBF%Bflx_temp_dz(:,:,k), G, &
+                    tmp_scale = GV%H_to_kg_m2*US%Z_to_m**2*US%s_to_T**3)
```

Comment:
As described in the MOM6 style guide at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#white-space, we strongly discourage the use of white space around an `=` when it is used to assign an optional argument using its name.  So line 148 should be `tmp_scale=GV%H_to_kg_m2*US%Z_to_m**2*US%s_to_T**3)`, with similar comments applying to lines 137 (`dz=dz`), 161, 189, 190, 201, 216, 217, 243, 244, 255, 270, 271, 282, 298, 309, 316, 327, 357, 372, 373, 384, 399, 400, 411,  and other places later in the code.  We have found this soft convention to be very helpful for searching through the code for actual assignments, and would prefer that it is resepected.

Reply:
Done, this is fixed in the latest commits.